packages/edit-site/src/components/block-editor/block-editor-provider/default-block-editor-provider.js

```diff
@@ -0,0 +1,135 @@
+/**
+ * WordPress dependencies
+ */
+import { useEntityBlockEditor } from '@wordpress/core-data';
+import { privateApis as blockEditorPrivateApis } from '@wordpress/block-editor';
+import { useSelect } from '@wordpress/data';
+import { useMemo } from '@wordpress/element';
+import { createBlock } from '@wordpress/blocks';
+
+/**
+ * Internal dependencies
+ */
+import { store as editSiteStore } from '../../../store';
+import { unlock } from '../../../lock-unlock';
+import useSiteEditorSettings from '../use-site-editor-settings';
+import { PAGE_CONTENT_BLOCK_TYPES } from '../../page-content-focus-manager/constants';
+
+const { ExperimentalBlockEditorProvider } = unlock( blockEditorPrivateApis );
+
+const noop = () => {};
+
+/**
+ * The default block editor provider for the site editor. Typically used when
+ * the post type is `'wp_template_part'` or `'wp_template'` and allows editing
+ * of the template and its nested entities.
+ *
+ * If the page content focus type is `'hideTemplate'`, the provider will provide
+ * a set of "ghost" blocks that mimick the look and feel of the post editor and
+ * allow editing of the page content only.
+ *
+ * @param {Object}    props
+ * @param {WPElement} props.children
+ */
+export default function DefaultBlockEditorProvider( { children } ) {
+	const settings = useSiteEditorSettings();
+
+	const { templateType, pageContentFocusType } = useSelect( ( select ) => {
+		const { getEditedPostType, getPageContentFocusType } =
+			select( editSiteStore );
+
+		return {
+			templateType: getEditedPostType(),
+			pageContentFocusType: getPageContentFocusType(),
+		};
+	}, [] );
+
+	const [ blocks, onInput, onChange ] = useEntityBlockEditor(
+		'postType',
+		templateType
+	);
+	const isPageContentFocused = pageContentFocusType === 'hideTemplate';
+	const pageContentBlock = usePageContentBlocks(
+		blocks,
+		isPageContentFocused
+	);
+	return (
+		<ExperimentalBlockEditorProvider
+			settings={ settings }
+			value={
+				isPageContentFocused && pageContentBlock.length
+					? pageContentBlock
+					: blocks
+			}
+			onInput={ isPageContentFocused ? noop : onInput }
+			onChange={ isPageContentFocused ? noop : onChange }
+			useSubRegistry={ false }
+		>
+			{ children }
+		</ExperimentalBlockEditorProvider>
+	);
+}
+
+const identity = ( x ) => x;
```

Comment:
Probably can ditch this since it's only used once internally and always with a `transform()`